experiment/scheduler.py

```diff
@@ -188,34 +187,17 @@ class TrialInstanceManager:  # pylint: disable=too-many-instance-attributes
     handle_preempted_trials method should be called in the scheduling loop.
     See the docstring for handle_preempted_trials for how it works.
     """
-    # Hard limit on the number of nonpreemptibles we will use. This bounds
```

Comment:
If switching to preemptibles earlier should be as cost effective as I'm led to beleive why do we need to remove these bounds on nonpreemptibles fraction and MAX_NONPREEMPTIBLES.

Reply:
So, these non premptibles bounds are bad, as it would mean we don't create trial at all if can_start_nonpreemptible is false. Keeping a simple logic feels better here, only create preemptibles the first day, and on second day, create nonpreemptibles whatever cost it is. It is always just a hundred or so, that might be needed on second day, so delaying the experiments for 3-4 days does not feel right here.

Reply:
I don't agree that they are bad. And we can simplify having them by just preventing creation of more than 10% nonpreemptibles.
But it's good to have this limit so that we don't end up doing a fully nonpreemptible experiment in rare cases

Reply:
It is complicating the code for unseen rare circumstance. I have never seen preemptible vms not available. 
As per old code, if 10% non-preemptible limit is reached, we don't create any more vms, that does not seem right either. We dont fall back to create preemptibles in that case.

Reply:
It seems right to me since this is such a rare event it's better to fail and let developer know than to try to handle it and waste a ton of money. I understand the event is rare but I think it's a good idea to defend against it.